### PR TITLE
array_str()

### LINKED ISSUE #11961: array_str
working on (#8431-> #17801)

### PR DESCRIPTION
**Pull request by Sarvesh-Kesharwani**
Closes #11961

Added array_str() and its test function:
1) for handing supress_small parameter.
2) to check the default parameter state of function.

### CODE REVIEW

File: ivy/functional/frontends/jax/numpy/creation.py [hunk 1]
```diff
@@ -182,3 +182,28 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
 @to_ivy_arrays_and_back
 def single(x):
     return ivy.astype(x, ivy.float32)
+
+
+@to_ivy_arrays_and_back
+def array_str(a, max_line_width=None, precision=None, suppress_small=False):
+    if precision is None:
+        # TODO: need implementation of get_printoptions() functional API
+        precision = ivy.get_printoptions()['precision']
```
Comment (juliagsy) [inline ivy/functional/frontends/jax/numpy/creation.py:191]: we have `ivy.array_significant_figures()` as an equivalent

File: ivy/functional/frontends/jax/numpy/creation.py [hunk 2]
```diff
@@ -182,3 +182,28 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
...
+            return 0
+        else:
+            return round(ele, precision_)
+
+    a = list(map(round_suppress, a, precision))
+    a = str(a)
+    
+    if max_line_width is None:
+        # TODO: need implementation of get_printoptions() functional API
+        max_line_width = ivy.get_printoptions()['linewidth']
```
Comment (juliagsy) [inline ivy/functional/frontends/jax/numpy/creation.py:205]: we don't have a function for this yet but you can hardcode it as `75` for the time being!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_creation.py [hunk 3]
```diff
@@ -755,3 +755,37 @@ def test_jax_numpy_single(
...
+    on_device,
+):
+    input_dtype, _, fill_value, dtype = input_fill_dtype
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        shape=shape,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_creation.py:787]: other than the default parameters for configuring the testing, the input parameters for testing the function should match the function

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Great work so far! I've commented the involved Ivy API below, thanks for the hard work!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by juliagsy** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**